pkg/platform/kube/functionres/lazy.go

```diff
@@ -2219,6 +2225,24 @@ func (lc *lazyClient) getFunctionVolumeAndMounts(ctx context.Context,
 
 				configVolume.Volume.FlexVolume.Options["subPath"] = subPath
 			}
+
+			// add secret ref to the volume if access key is scrubbed
+			accessKey, accessKeyExists := configVolume.Volume.FlexVolume.Options["accessKey"]
+			if accessKeyExists && strings.HasPrefix(accessKey, functionconfig.ReferencePrefix) {
+
+				// get the flex volume secret name
+				secretName, err := lc.getFlexVolumeSecretName(ctx, function, configVolume.Volume.Name)
+				if err != nil {
+					lc.logger.WarnWithCtx(ctx, "Failed to get flex volume secret name. Ignoring volume",
+						"err", err)
+					continue
```

Comment:
```suggestion
					lc.logger.WarnWithCtx(ctx, 
					"Failed to get flex volume secret name. Ignoring volume",
						"err", err)
					continue
```

Reply:
👀 ignoring a volume? how can that be safe or normal to the user? his function would by deployed successfully while no volumes would be in place. I suggest failing here while explaining what went wrong